### PR TITLE
[debezium] Add `NewIterator()` to `Adapter` interface

### PR DESCRIPTION
**Pull request by nathan-artie**

### CODE REVIEW

File: lib/debezium/transformer.go [hunk 1]
```diff
@@ -11,21 +11,29 @@ import (
 	"github.com/artie-labs/reader/lib"
 )
 
+type Row = map[string]any
```
Comment (Tang8330) [inline lib/debezium/transformer.go:14]:
```go
type Row map[string]any
``` 

instead?

Comment (nathan-artie) [reply lib/debezium/transformer.go:14]: Are they equivalent or does that doe something different?

Comment (nathan-artie) [reply lib/debezium/transformer.go]: Hmm that doesn't seem to work.

Comment (Tang8330) [reply lib/debezium/transformer.go:14]: oh you are trying to do an alias, I see.

**Review by Tang8330 — APPROVED** [review]

**Pull request merged**